pkg/delete/delete.go

```diff
@@ -38,91 +35,78 @@ import (
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/delete/pointer"
 )
 
-type ClientSet struct {
-	Classic    client.ConfigClient
-	Settings   client.SettingsClient
-	Automation client.AutomationClient
-	Buckets    client.BucketClient
-	Documents  client.DocumentClient
-}
-
 type configurationType = string
 
 // DeleteEntries is a map of configuration type to slice of delete pointers
 type DeleteEntries = map[configurationType][]pointer.DeletePointer
 
 // Configs removes all given entriesToDelete from the Dynatrace environment the given client connects to
-func Configs(ctx context.Context, clients ClientSet, _ api.APIs, automationResources map[string]config.AutomationResource, entriesToDelete DeleteEntries) error {
-	copiedDeleteEntries := make(DeleteEntries)
-	for k, v := range entriesToDelete {
-		copiedDeleteEntries[k] = v
+func Configs(ctx context.Context, clients client.ClientSet, entriesToDelete DeleteEntries) error {
+	remainingEntriesToDelete, errCount := deleteAutomationConfigs(ctx, clients.AutClient, entriesToDelete)
+
+	//  Dashboard share settings cannot be deleted
+	if _, ok := remainingEntriesToDelete[api.DashboardShareSettings]; ok {
+		log.Warn("Classic config of type %s cannot be deleted. Note, that they can be removed by deleting the associated dashboard.", api.DashboardShareSettings)
+		delete(remainingEntriesToDelete, api.DashboardShareSettings)
+
+	}
+
+	// Delete rest of config types
+	for t, entries := range remainingEntriesToDelete {
+		if err := deleteOtherConfigs(ctx, clients, t, entries); err != nil {
```

Comment:
Is `deleteConfig` sounds better than `deleteOthereConfigs`?
`deleteAutmoationConfig` is a special case, but this is standard behavior.